examples/compose/plot_column_transformer_mixed_types.py

```diff
@@ -68,16 +68,52 @@
 
 # Append classifier to preprocessing pipeline.
 # Now we have a full prediction pipeline.
-clf = Pipeline(steps=[('preprocessor', preprocessor),
-                      ('classifier', LogisticRegression(solver='lbfgs'))])
+pipeline = Pipeline(steps=[('preprocessor', preprocessor),
+                           ('classifier', LogisticRegression(solver='lbfgs'))])
 
 X = data.drop('survived', axis=1)
 y = data['survived']
 
 X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
 
-clf.fit(X_train, y_train)
-print("model score: %.3f" % clf.score(X_test, y_test))
+pipeline.fit(X_train, y_train)
+print("model score: %.3f" % pipeline.score(X_test, y_test))
+
+
+###############################################################################
+# Introspecting the coefficients values of the classifier
+###############################################################################
+# The coefficients of the final classification step of the pipeline gives an
+# idea how each feature impacts the likelihood of survival assuming that the
+# usual linear model assumptions hold (uncorrelated features, linear
+# separability, homoschedastic and normally distributed errors...) which we do
+# not verify in this example.
+#
+# To get error bars we perform cross-validation and compute the mean and
+# standard deviation for each coefficient accross CV splits. Because we use a
+# standard scaler on the numerical features, the coefficient weights gives us
+# an idea on how much the log odds of surviving are impacted by a change in
+# this dimension contrasted to the mean. Note that the categorical features
+# here are overspecified which makes it slightly harder to interpret because of
+# the information redundancy.
+#
+# We can see that the linear model coefficients are in agreement with the
+# historical reports: people in higher classes and therefore in the upper decks
+# were first to access the lifeboats, and often, priority was given to women
+# and children.
+
+import matplotlib.pyplot as plt
+from sklearn.model_selection import cross_validate
+
+cv_results = cross_validate(pipeline, X_train, y_train, cv=10,
+                            return_estimator=True)
+cv_coefs = np.concatenate([cv_pipeline.named_steps["classifier"].coef_
```

Comment:
As mentioned IRL, I think that we could change the "_last_estimator" of the pipeline to "last_estimator_" and use it here so that this code snippet (which is a very common one) is independent of the name given to the last step:
```
cv_pipeline.last_estimator_.coef_
```
and below
```
pipeline.last_estimator_.input_features_
```

Reply:
If we do so, I guess `last_step_` would be better so that we don't have to deal with the last step potentially being only a _transformer_ and so on.

Reply:
A transformer is regarded as an estimator. I think `last_step_` is sufficient and succinct, though.

Reply:
we now have [-1] which I think is fine.